python/revng/cli/_commands/fetch_debuginfo/__init__.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+
+import pefile
+from elftools.common.exceptions import ELFError
+from elftools.elf.elffile import ELFFile
+
+from revng.cli.commands_registry import Command, CommandsRegistry, Options
+from revng.cli.support import log_error
+
+from .common import log
+from .elf import fetch_dwarf
+from .pe import fetch_pdb
+
+default_elf_debug_server = "https://debuginfod.elfutils.org/"
+microsoft_symbol_server_url = "http://msdl.microsoft.com/download/symbols/"
+
+
+# For now, the result of the tool is a path of the debug info file printed on the stdout.
+def log_result(verbose, file_path=""):
+    if file_path != "":
+        log(f"Result: {file_path}", verbose)
+    else:
+        log("Result: No debug info found", verbose)
+
+
+class FetchDebugInfoCommand(Command):
+    def __init__(self):
+        super().__init__(("model", "fetch-debuginfo"), "Fetch Debugging Information.")
+
+    def register_arguments(self, parser):
+        parser.add_argument("input", help="The input file.")
+        parser.add_argument(
+            "urls",
+            help="List of URLs where to try finding the debug symbols.",
+            nargs="*",
+            default=[],
+        )
+
+    def run(self, options: Options):
+        args = options.parsed_args
+        self.verbose = args.verbose
+
+        if not os.path.exists(args.input):
+            log_error("Could not find " + args.input)
+            return 1
+
+        path = os.path.abspath(args.input)
+        with open(args.input, "rb") as f:
+            try:
+                the_elffile = ELFFile(f)
+                if not args.urls:
+                    args.urls.append(default_elf_debug_server)
+
+                result = fetch_dwarf(the_elffile, path, args.urls, self.verbose)
+            except ELFError as elf_error:
+                if str(elf_error) == "Magic number does not match":
```

Comment:
* Does `ELFError` has `.message` or something like that?
* Consider either dropping this check and just go to pefile in case any `ELFError` arises, or maybe manually check for the ELF header. Manually comparing error message seems a bit fragile.

Reply:
I will check it manually. I agree.